plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py

```diff
@@ -1,177 +1,169 @@
-    IntelType.FILENAME_MD5: ("FILE_NAME", "FILE_HASH"),
-    IntelType.SHA1: "FILE_HASH",
-    IntelType.FILENAME_SHA1: ("FILE_NAME", "FILE_HASH"),
-    IntelType.SHA256: "FILE_HASH",
-    IntelType.FILENAME_SHA256: ("FILE_NAME", "FILE_HASH"),
-    IntelType.X509FINGERPRINTSHA1: "CERT_HASH",
-    IntelType.PDB: "FILE_NAME",
-    IntelType.AUTHENTIHASH: "FILE_HASH",
-    IntelType.SSDEEP: "FILE_HASH",
-    IntelType.IMPHASH: "FILE_HASH",
-    IntelType.PEHASH: "FILE_HASH",
-    IntelType.IMPFUZZY: "FILE_HASH",
-    IntelType.SHA224: "FILE_HASH",
-    IntelType.SHA384: "FILE_HASH",
-    IntelType.SHA512: "FILE_HASH",
-    IntelType.SHA512_224: "FILE_HASH",
-    IntelType.SHA512_256: "FILE_HASH",
-    IntelType.TLSH: "FILE_HASH",
-    IntelType.CDHASH: "FILE_HASH",
-    IntelType.FILENAME_AUTHENTIHASH: ("FILE_NAME", "FILE_HASH"),
-    IntelType.FILENAME_SSDEEP: ("FILE_NAME", "FILE_HASH"),
-    IntelType.FILENAME_IMPHASH: ("FILE_NAME", "FILE_HASH"),
-    IntelType.FILENAME_PEHASH: ("FILE_NAME", "FILE_HASH"),
-    IntelType.FILENAME_IMPFUZZY: ("FILE_NAME", "FILE_HASH"),
-    IntelType.FILENAME_SHA224: ("FILE_NAME", "FILE_HASH"),
-    IntelType.FILENAME_SHA384: ("FILE_NAME", "FILE_HASH"),
-    IntelType.FILENAME_SHA512: ("FILE_NAME", "FILE_HASH"),
-    IntelType.FILENAME_SHA512_224: ("FILE_NAME", "FILE_HASH"),
-    IntelType.FILENAME_SHA512_256: ("FILE_NAME", "FILE_HASH"),
-    IntelType.FILENAME_TLSH: ("FILE_NAME", "FILE_HASH"),
-}
+from typing import Union
 
-from_zeek_intel = {
-    "ADDR": IntelType.IPSRC,
-    "EMAIL": IntelType.EMAILSRC,
-    "FILE_NAME": IntelType.FILENAME,
-    "DOMAIN": IntelType.DOMAIN,
-    "URL": IntelType.URL,
-    "SOFTWARE": IntelType.USERAGENT,
-    "FILE_HASH": IntelType.MD5,
+# See the documentation for the Zeek INTEL framework [1] and STIX-2 cyber
+# observable objects [2]
+# [1] https://docs.zeek.org/en/stable/scripts/base/frameworks/intel/main.zeek.html#type-Intel::Type
+# [2] https://docs.oasis-open.org/cti/stix/v2.1/cs01/stix-v2.1-cs01.html#_mlbmudhl16lr
+zeek_intel_type_map = {
+    "domain-name:value": "DOMAIN",
+    "email-addr:value": "EMAIL",
+    "file:name": "FILE_NAME",
+    "file:hashes.MD5": "FILE_HASH",
+    "file:hashes.'SHA-1'": "FILE_HASH",
+    "file:hashes.'SHA-256'": "FILE_HASH",
+    "file:hashes.'SHA-512'": "FILE_HASH",
+    "file:hashes.'SHA3-256'": "FILE_HASH",
+    "file:hashes.'SHA3-512'": "FILE_HASH",
+    "file:hashes.SSDEEP": "FILE_HASH",
+    "file:hashes.TLSH": "FILE_HASH",
+    "ipv4-addr:value": "ADDR",
+    "ipv6-addr:value": "ADDR",
+    "software:name": "SOFTWARE",
+    "url:value": "URL",
+    "user:user_id": "USER_NAME",
+    "user:account_login": "USER_NAME",
+    "x509-certificate:hashes.'SHA-1'": "CERT_HASH",  # Zeek only supports SHA-1
 }
 
 
-def map_management_message(broker_data, module_namespace):
-    """Maps a management message to an actionable instruction for threatbus.
+def map_management_message(
+    broker_data, module_namespace: str
```

Comment:
Why does this function not get a handle to the logger? Is the volume of messages too high?

Reply:
No the logvolume would be very low. I changed the code to pass a logger instance to that function and made the function log unexpected subscription messages.